### PR TITLE
Cache key fix

### PR DESCRIPTION
**Pull request by novikor**
### Description (*)
Optimized cache usage for TopMenu block.
Previously, it was cached for ALL pages in order to highlight current category in navigation bar, however since 2.2 it was replaced with JS solution.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @novikor. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by orlangur — CHANGES_REQUESTED** [review]
@novikor it needs to de done for 2.2 as backport of https://github.com/magento/magento2/commit/61d81a8e07b627eb7f3850529767e1427461f52b#diff-454601ea9afd086413ac90923ae9ccaa and https://github.com/magento/magento2/commit/fcb3d675ea3df97f8935cc5a560a396ac9ae7db2

**Comment by alejagapatrick10** [comment]
Thanks

**Comment by alejagapatrick10** [comment]
NY_license.pdf
<https://drive.google.com/file/d/1GI9XvBeBVBG96h2Xn7xrl9wnqO5DaLFR/view?usp=drivesdk>



Noong Lin, Nob 4, 2018, 02:15 alejagapatrick10 Isinulat ni <
notifications@github.com ang:

**Comment by novikor** [comment]
@orlangur, will be done. Thank you for review.

**Comment by novikor** [comment]
@orlangur , please check

**Comment by orlangur** [comment]
@novikor commits are not properly authored yet. Please cherry-pick them and force push to the same branch.

**Comment by novikor** [comment]
@orlangur , please review again

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-3557](https://jira.corp.magento.com/browse/ENGCOM-3557) has been created to process this Pull Request

File: app/code/Magento/Theme/Block/Html/Topmenu.php [hunk 1]
```diff
@@ -361,19 +361,6 @@ public function getIdentities()
         return $this->identities;
     }
 
-    /**
-     * Get cache key informative items
-     *
-     * @return array
-     * @since 100.1.0
-     */
-    public function getCacheKeyInfo()
```
Comment (sidolov) [inline app/code/Magento/Theme/Block/Html/Topmenu.php:370]: Unfortunately, we can't remove the public methods according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

**Review by sidolov — CHANGES_REQUESTED** [review]

File: app/code/Magento/Theme/Block/Html/Topmenu.php (hunk 1 above)
Comment (orlangur) [reply app/code/Magento/Theme/Block/Html/Topmenu.php:370]: @sidolov I don't think it's applicable. We have a parent method `parent::getCacheKeyInfo`, isn't it?

**Review by orlangur — CHANGES_REQUESTED** [review]
@novikor there are still unnecessary merge commits.

Could you please force push a branch with clean history?

Let's wait for reply of @sidolov regarding removal first.

**Comment by sidolov** [comment]
Hi @novikor , @orlangur , looks like the same fix was delivered in commit d30ae920ac10b1382a66763a2e0222af4f5dd731 and delkivered to the mainline. 
I'm closing this PR as not relevant. 
Thank you for the contribution!

**Comment by orlangur** [comment]
@sidolov thanks! Would be nice if internal contributors will do proper cherry-picks during backport as well ;)

**Comment by sidolov** [comment]
@orlangur I have raised this problem in our internal team and we will try to avoid such situations in the future! Thank you for the input!

**Pull request closed without merging**